pkgs/misc/rkdeveloptool-pine64/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchurl
+, meson
+, pkg-config
+, libusb1
+, scdoc
+, ninja
+, cmake
+}:
+
+let
+  rev = "cce7d2a5c4efd4e7727c440868141229354b327b";
+in
+stdenv.mkDerivation {
+  pname = "rkdeveloptool";
+  version = "unstable-2021-09-04";
+
+  src = fetchurl {
+    url = "https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/-/archive/${rev}/rkdeveloptool-${rev}.tar.gz";
+    sha256 = "sha256-u/x1Y1zZ19SYwNLVAvpqjH247RijyDJ1HTDWIsmqlFk=";
+  };
+
+  postPatch = ''
+    substituteInPlace meson.build --replace \
+      "udev_rules_dir = udev.get_pkgconfig_variable('udevdir') + '/rules.d'" \
+      "udev_rules_dir = '$out/lib/udev'"
+  '';
+
+  nativeBuildInputs = [ meson ninja cmake pkg-config scdoc ];
+
+  buildInputs = [ libusb1 ];
+
+  meta =
+    let
+      inherit (lib) maintainers;
+    in
+    {
+      homepage = "https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/";
+      description = "A tool from Rockchip to communicate with Rockusb devices (pine64 fork)";
+      license = lib.licenses.gpl2;
```

Comment:
Unfree? Unclear freedom?

 - https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/-/blob/cce7d2a5c4efd4e7727c440868141229354b327b/Property.hpp#L27-30

Files with license info in the repo (SPDX header here) lists it as GPL-2.0+

 - https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/-/blob/cce7d2a5c4efd4e7727c440868141229354b327b/RKBoot.cpp#L5
 - https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/-/blob/cce7d2a5c4efd4e7727c440868141229354b327b/RKComm.cpp#L5
 - https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/-/blob/cce7d2a5c4efd4e7727c440868141229354b327b/RKDevice.cpp#L5
 - https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/-/blob/cce7d2a5c4efd4e7727c440868141229354b327b/RKImage.cpp#L5
 - https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/-/blob/cce7d2a5c4efd4e7727c440868141229354b327b/RKLog.cpp#L5
 - https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/-/blob/cce7d2a5c4efd4e7727c440868141229354b327b/RKScan.cpp#L5
 - https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/-/blob/cce7d2a5c4efd4e7727c440868141229354b327b/crc.cpp#L5
 - https://gitlab.com/pine64-org/quartz-bsp/rkdeveloptool/-/blob/cce7d2a5c4efd4e7727c440868141229354b327b/main.cpp#L5

Reply:
The original rkdeveloptool is marked as GPL-2 in nixpkgs (from which most of these files are inherited).
I didn't look much further than that considering that this is a fork, and therefore by extension, also licensed under the same terms.